app/src/main/java/com/gh4a/fragment/RepositoryFragment.java

```diff
@@ -72,44 +67,13 @@ public static RepositoryFragment newInstance(Repository repository, String ref)
     private String mRef;
     private HttpImageGetter mImageGetter;
 
-    private final LoaderCallbacks<String> mReadmeCallback = new LoaderCallbacks<String>(this) {
-        @Override
-        protected Loader<LoaderResult<String>> onCreateLoader() {
-            return new ReadmeLoader(getActivity(), mRepository.getOwner().getLogin(),
-                    mRepository.getName(), StringUtils.isBlank(mRef) ? mRepository.getDefaultBranch() : mRef);
-        }
-        @Override
-        protected void onResultReady(String result) {
-            TextView readmeView = (TextView) mContentView.findViewById(R.id.readme);
-            View progress = mContentView.findViewById(R.id.pb_readme);
-            AsyncTaskCompat.executeParallel(new FillReadmeTask(
-                    mRepository.getId(), readmeView, progress, mImageGetter), result);
-        }
-    };
-
-    private final LoaderCallbacks<Integer> mPullRequestsCallback = new LoaderCallbacks<Integer>(this) {
-        @Override
-        protected Loader<LoaderResult<Integer>> onCreateLoader() {
-            return new PullRequestCountLoader(getActivity(), mRepository, ApiHelpers.IssueState.OPEN);
-        }
-
-        @Override
-        protected void onResultReady(Integer result) {
-            View v = getView();
-            v.findViewById(R.id.issues_progress).setVisibility(View.GONE);
-            v.findViewById(R.id.pull_requests_progress).setVisibility(View.GONE);
-
-            TextView tvIssuesCount = (TextView) mContentView.findViewById(R.id.tv_issues_count);
-            tvIssuesCount.setText(String.valueOf(mRepository.getOpenIssues() - result));
-
-            TextView tvPullRequestsCountView = (TextView) v.findViewById(R.id.tv_pull_requests_count);
-            tvPullRequestsCountView.setText(String.valueOf(result));
-        }
-    };
+    private Observable<String> mLoadReadme;
+    private Disposable[] mSubscriptions;
 
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mSubscriptions = new Disposable[2];
```

Comment:
Maybe it would be better to use some sort of map? That way we would avoid potential out of bounds exceptions when someone adds new subscription but doesn't update that declaration.

Reply:
It was temporary, but nice consideration. Ideally, in order to share disposables unsubscription to avoid memory leaks, this logic should be shared over all Activities and Fragments, since every one of them could make use of an Observable, there's also **CompoundSubscription** object in **rx** that may help ...